packages/x-components/src/components/scroll/use-scroll.ts

```diff
@@ -113,9 +113,9 @@ export function useScroll(
    */
   const storeScrollData = () => {
     if (scrollEl.value) {
-      currentPosition.value = scrollEl.value.scrollTop;
       scrollHeight.value = scrollEl.value.scrollHeight;
       clientHeight.value = scrollEl.value.clientHeight;
+      currentPosition.value = scrollEl.value.scrollTop;
```

Comment:
This change makes sense in terms of watch callbacks. I was debugging the same example with this composable and the mixin and the side effect seems to be the order of reactivity execution regarding `scrollDirection` (as you explain very well in the PR description).

I don't like the `isClientHeightChanging` concept itself, and I think is the key part of the issue and side effects. It could be refactored to handle the reactivity and the side effects in a more controlled way, but for the time being, this solution is easy and works so perfect.

WDYT to add a unit test sheet for it and/or add a test in the scroll e2e? 😅

Reply:
Added a specific e2e to check for the scrolling direction 👍